src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,97 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.TalonFXControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.utils.DeadbandProximity;
+
+import static frc.robot.Constants.Conveyor.*;
+
+public class Conveyor extends SubsystemBase {
+    private static int balls = Constants.Conveyor.INITIAL_BALLS_AMOUNT;
+
+    private final TalonFX motor = new TalonFX(Ports.Conveyor.MOTOR);
```

Comment:
Why is `balls` static? You have only one instance of this class anyway.

Reply:
I use it because I don't think that the `balls` is specific to the conveyor, the balls is being used in most of the subsystem of the robot.
If you want, I can move it to the `Robot` class, because the balls are not belonged to a specific subsystem...

Reply:
Exactly

Reply:
OK, so instead, export a method from the `Conveyor` class which says whether the amount of Power Cells is at the allowed maximum.

Reply:
first of all, you didn't provide any reason for not using `synchronized` or `static`.
Second of all, No matter what, I still need to update the amount of balls, so I need to keep the `synchronized` keyword.

Reply:
> first of all, you didn't provide any reason for not using `synchronized` or `static`.

That's on you. You add code, you need to justify its existence.
> No matter what, I still need to update the amount of balls

True, but how does that imply
> I need to keep the `synchronized` keyword

?
You need `synchronized` when you write multi-threaded code (which you don't), and `static` when you have multiple instances of a class (which you don't, as subsystems are [singletons](https://en.wikipedia.org/wiki/Singleton_pattern)).

Reply:
> That's on you. You add code, you need to justify its existence.

And you need to show why we don't need to update the value.
I will delete (some of) the `synchronized`, but not because what you said, because you didn't give me any reason that I'm wrong.

> `static` when you have multiple instances of a class

not only, when the variable is not (directly) related to the class. The `balls` variable isn't related only to the conveyor.
I told you, if you want, I can move it to the Robot class

Reply:
> but not because what you said

At first, I thought whether I should feel offended.
> I will delete (some of) the `synchronized`, but not because what you said, because you didn't give me any reason that I'm wrong.

But I did (https://github.com/Galaxia5987/robot-2021/pull/9#discussion_r566845009):
> You need `synchronized` when you write multi-threaded code (which you don't)
<br>

> And you need to show why we don't need to update the value.

Again, I've never said you don't need to update the amount of Power Cells the conveyor holds:
> > No matter what, I still need to update the amount of balls
>
> True
<br>

> not only, when the variable is not (directly) related to the class

Could you please provide an example? I've never considered that to be a good design idea. A class should only contain members it needs.
> The `balls` variable isn't related only to the conveyor. I told you, if you want, I can move it to the Robot class

You're right, I rejected that idea at first. After all, the conveyor stores the Power Cells, and the only subsystem other than it that cares about the number of them is, maybe, the intake. But that's more than one subsystem, which justifies moving it to the `Robot` class…
_However_, while writing the above paragraph, I thought again it's not a good idea – by storing the counter in the `Robot` class, you let every other component in the code change it. But it should only be modified by the conveyor because it holds the sensors responsible to count the Power Cells.